src/exchange/sockets.ts

```diff
@@ -209,4 +223,31 @@ export class Sockets {
   ) => {
     this.socketInstance.on(SOCKET_EVENTS.AccountDataUpdateKey, cb);
   };
+
+
+  async onDisconnect(): Promise<void> {
+    this.socketInstance.on("disconnect", async () => {
+      console.log('Disconnected From Socket Server');
+      if ('disconnect' in Sockets.callbacks && typeof Sockets.callbacks['disconnect'] === 'function') {
+        await Sockets.callbacks['disconnect']();
+      }
+    });
+
+  }
+
+  async onConnect(): Promise<void> {
+    this.socketInstance.on("connect", async () => {
+      console.log('Connected To Socket Server');
+      if ('connect' in Sockets.callbacks && typeof Sockets.callbacks['connect'] === 'function') {
+        // Add 10 seconds sleep using setTimeout
+        setTimeout(async () => {
```

Comment:
Why have we induced this delay?

Reply:
This delay is to give time for socket connection to actually be established because it was noticed that even after connect listener is triggered, connection actually is not built immediately due to asynchronous nature of the calls inside socket.io.

Reply:
And to add, I tried lesser numbers for timeout as well but it wasnt until 10 seconds later I was able to see the connection finally.